### PR TITLE
Added minMuleVersion check for Connectors

### PR DESCRIPTION
**Pull request by giordanolucas**

### CODE REVIEW

File: src/main/java/org/mule/tools/devkit/sonar/checks/java/MinMuleVersionCheck.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    public static final String KEY = "min-mule-version";
+
+    @Override
+    protected void verifyConnector(ClassTree classTree, IdentifierTree connectorAnnotation) {
+
+        for (AnnotationTree annotationTree : Iterables.filter(classTree.modifiers().annotations(), ClassParserUtils.ANNOTATION_TREE_PREDICATE)) {
+            if (ClassParserUtils.is(annotationTree, "org.mule.api.annotations.Connector")) {
+                for(ExpressionTree expression : annotationTree.arguments()){
+                    if(expression.kind().name().equals("ASSIGNMENT")){
+                        if(((IdentifierTreeImpl) ((AssignmentExpressionTreeImpl) expression).variable()).name().equals("minMuleVersion")){
```
Comment (urbanusjam) [inline src/main/java/org/mule/tools/devkit/sonar/checks/java/MinMuleVersionCheck.java]: Missing validation for empty minMuleVersion.

File: src/main/java/org/mule/tools/devkit/sonar/checks/java/MinMuleVersionCheck.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.java.model.expression.AssignmentExpressionTreeImpl;
+import org.sonar.java.model.expression.IdentifierTreeImpl;
+import org.sonar.plugins.java.api.tree.*;
+import org.sonar.squidbridge.annotations.ActivatedByDefault;
+
+import java.util.List;
+
+@Rule(key = MinMuleVersionCheck.KEY, name = "Connectors should specify a minMuleVersion", description = "The minMuleVersion should be specified on the @Connector annotation.", priority = Priority.CRITICAL, tags = { "connector-certification" })
```
Comment (urbanusjam) [inline src/main/java/org/mule/tools/devkit/sonar/checks/java/MinMuleVersionCheck.java]: Minor fixes for the description: The **'minMuleVersion'** should be specified **in** the @Connector annotation.

File: src/main/java/org/mule/tools/devkit/sonar/checks/java/MinMuleVersionCheck.java (hunk 1 above)
Comment (giordanolucas) [inline src/main/java/org/mule/tools/devkit/sonar/checks/java/MinMuleVersionCheck.java]: Fixed!

File: src/main/java/org/mule/tools/devkit/sonar/checks/java/MinMuleVersionCheck.java (hunk 2 above)
Comment (giordanolucas) [inline src/main/java/org/mule/tools/devkit/sonar/checks/java/MinMuleVersionCheck.java]: Fixed!

**Pull request merged**